fix(storybook): Fix theme error in Docs tab

vuluongj20 opened the pull request:
The Docs tab of some stories in Storybook is blank with a theme parsing error:
<img width="1998" alt="image" src="https://user-images.githubusercontent.com/44172267/169584267-3754436d-3e12-4409-acb0-53ebc442c1ca.png">

Turns out this is from my previous commit – d184f66a335e7616fc9247cd12b3d3f707eb3ce0 – that replaced the `background` property from Storybook's default theme object with the one from our own theme. Storybook's default value is an object, while ours is a string, hence the parsing error. The commit was submitted to fix an issue where our custom components (the ones in `docs-ui/components`) expect a string value out of the `background` property, not an object, so I chose to overwrite `background` with our own string value. 

The better solution, adopted here, that avoids errors in either Storybook's components or our custom components is to just keep Storybook's default object value (hence letting Storybook's default components to work normally), while providing an additional `bg` property for consumption by our custom components in `docs-ui/components`.

vuluongj20 commented on static/app/utils/theme.tsx:
```diff
@@ -865,7 +865,16 @@ type MyTheme = Theme;
 declare module '@emotion/react' {
   // eslint-disable-next-line @typescript-eslint/no-shadow
-  export interface Theme extends MyTheme {}
+  export interface Theme extends MyTheme {
+    /**
+     * [For Storybook only, do not use inside the app.]
+     *
+     * Substitute for the "background" property in Storybook components
+     * (anything inside docs-ui), since Storybook overrides "background" with
+     * its own object value.
+     */
+    bg?: string;
+  }
```
Extending Theme here so we don't get typescript warnings when using `theme.bg` in `docs-ui`.

billyvg replied:
I would declare this somewhere in our storybook files, otherwise it will be confusing for people working on the sentry app.

billyvg commented on docs-ui/storybook/preview.tsx:
```diff
@@ -44,6 +43,7 @@ addDecorator(WithThemeStory);
 const WithThemeDocs: DecoratorFn = ({children, context}) => {
   const isDark = useDarkMode();
   const currentTheme = isDark ? darkTheme : lightTheme;
+  currentTheme.bg = currentTheme.background;
```
Can we call it `docsBackground` maybe?

vuluongj20 commented:
@billyvg ahh good points! updated the color name to `docsBackground` and moved the Theme declaration to inside `docs-ui/`: https://github.com/getsentry/sentry/pull/34878/commits/15850a58e3aa34ea9c372c0a6b971fcc7f340ba0
